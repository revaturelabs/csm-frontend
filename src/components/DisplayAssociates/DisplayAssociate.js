import React, { useState } from 'react';
import { Accordion, Icon, Button } from 'semantic-ui-react';
import { useHistory } from 'react-router-dom';
import { useDispatch } from "react-redux";
import AssociateService from '../../services/associate.service'
import './DisplayAssociates.scss';

const DisplayAssociate = (props) => {
  const history = useHistory();
  const dispatch = useDispatch();
  const associateService = new AssociateService();
  const [assocActiveIndex, setAssocActiveIndex] = useState(-1);

  const handleClick = (e, titleProps) => {
    const { index } = titleProps
    const newIndex = assocActiveIndex === index ? -1 : index
    setAssocActiveIndex(newIndex)
  }

  const viewSwots = () => {
    const getAssociateInfo = async () => {
      let resp = await associateService.getSpiderInformation(props.associate.userID)
      dispatch({ type: 'setStartDate', date: new Date(new Date().setDate(new Date().getDate() - 14))})
      dispatch({ type: 'setEndDate', date: new Date(props.batchProDate)})
      dispatch({ type: 'updateAssociate', associate: resp.data })
      history.push('/viewSwots');
    }
    getAssociateInfo();
  }

  return (
    <Accordion styled className='associate-accordion'>
      <Accordion.Title
        active={assocActiveIndex === 0}
        index={0}
        onClick={handleClick}
        className='title'>
        <Icon name='dropdown' />
        <span className='id'>{props.associate.name}</span>
        <span className='name'>{props.associate.userID}</span>
        <span className='batch'>{props.batchName}</span>
        <span className='pro_date'>Promoted: {props.batchProDate}</span>
      </Accordion.Title>
      <Accordion.Content active={assocActiveIndex === 0}>
        {/* placeholder for the evaluation charts */}
        <span><Button onClick={viewSwots}>View SWOTs</Button></span>
      </Accordion.Content>
    </Accordion>
  )
}

export default DisplayAssociate;
