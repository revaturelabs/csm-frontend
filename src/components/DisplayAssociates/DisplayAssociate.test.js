import React from 'react';
import DisplayAssociate from './DisplayAssociate';
import renderer from 'react-test-renderer';

test('component renders', () => {
    const testAssociate = {
        'ID':1,
        'Name':'TestName',
        'batch':'test-batch'
    }
    const component = renderer.create(
        <DisplayAssociate associate={testAssociate}/>
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
})