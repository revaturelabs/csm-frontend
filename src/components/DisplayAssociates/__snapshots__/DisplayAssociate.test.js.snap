// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component renders 1`] = `
<tr
  className=""
>
  <td
    className=""
  >
    1
  </td>
  <td
    className=""
  >
    TestName
  </td>
  <td
    className=""
  >
    test-batch
  </td>
  <div
    class="circular ui icon button"
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <i
      class="icon eye"
    />
  </div>
</tr>
`;
